src/apps/filters/h3.c

```diff
@@ -348,6 +361,328 @@ SUBCOMMAND(getIcosahedronFaces,
     return E_SUCCESS;
 }
 
+/// Traversal subcommands
+
+SUBCOMMAND(
+    gridDisk,
+    "Returns a JSON array of a H3 cells within 'k' steps of the origin cell") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    int k = 0;
+    Arg kArg = {.names = {"-k"},
+                .required = true,
+                .scanFormat = "%d",
+                .valueName = "distance",
+                .value = &k,
+                .helpText = "Maximum grid distance for the output set"};
+    Arg *args[] = {&gridDiskArg, &helpArg, &cellArg, &kArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    int64_t len = 0;
+    H3Error err = H3_EXPORT(maxGridDiskSize)(k, &len);
+    if (err) {
+        return err;
+    }
+    H3Index *out = calloc(len, sizeof(H3Index));
+    err = H3_EXPORT(gridDisk)(cell, k, out);
+    if (err) {
+        free(out);
+        return err;
+    }
+    // Since we don't know *actually* how many cells are in the output (usually
+    // the max, but sometimes not), we need to do a quick scan to figure out the
+    // true length in order to properly serialize to a JSON array
+    int64_t trueLen = 0;
+    for (int64_t i = 0; i < len; i++) {
+        if (out[i] != 0) {
+            trueLen++;
+        }
+    }
+    printf("[ ");
+    for (int64_t i = 0, j = 0; i < len; i++) {
+        if (out[i] != 0) {
+            j++;
+            printf("\"%" PRIx64 "\"%s", out[i], j == trueLen ? "" : ", ");
+        }
+    }
+    free(out);
+    printf(" ]\n");
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(
+    gridDiskDistances,
+    "Returns a JSON array of arrays of H3 cells, each array containing cells "
+    "'k' steps away from the origin cell, based on the outer array index") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    int k = 0;
+    Arg kArg = {.names = {"-k"},
+                .required = true,
+                .scanFormat = "%d",
+                .valueName = "distance",
+                .value = &k,
+                .helpText = "Maximum grid distance for the output set"};
+    Arg prettyArg = {
```

Comment:
FWIW, I'd skip - interested callers have access to plenty of other pretty-printing tools